src/hooks/useComments.js

```diff
@@ -1,18 +1,22 @@
 import { useEffect, useState } from "react";
 import { getArticleCommentsById } from "../api";
 
-const useComments = (id) => {
+const useComments = (id, limit) => {
     const [commentsData, setCommentsData] = useState([]);
+    const [totalComments, setTotalComments] = useState(0);
     const [isLoading, setIsLoading] = useState(true);
     
     useEffect(() => {
         setIsLoading(true);
-        getArticleCommentsById(id)
-        .then(data => setCommentsData(data))
+        getArticleCommentsById(id, limit)
+        .then(data => {
+            setCommentsData(data.comments);
+            setTotalComments(data.total_count);
+        })
         .then(() => setIsLoading(false));
```

Comment:
Nice loading pattern. You might want to consider using .finally() over .then() here